src/explore-education-statistics-frontend/src/modules/data-catalogue/DataCataloguePage.tsx

```diff
@@ -112,11 +113,24 @@ const DataCataloguePage: NextPage<Props> = ({
   const handleDownloadFormSubmit: DownloadFormSubmitHandler = async ({
     files,
   }) => {
-    if (!state.query.release) {
+    const { release } = state.query;
+
+    if (!release) {
       throw new Error('Release has not been selected');
     }
 
-    await downloadService.downloadFiles(state.query.release.id, files);
+    await downloadService.downloadFiles(release.id, files);
+
+    const availableFiles = state.subjects.map(subject => subject.file);
+    const selectedFilenames = files
+      .map(fileId => availableFiles.find(file => file.id === fileId)?.fileName)
+      .join(', ');
```

Comment:
I suppose there's the potential for something like `'file a, file b, , file d` if an `undefined` creeps in the result of the `find` but that should never happen?